raiden-dapp/src/services/raiden-service.ts

```diff
@@ -120,34 +121,25 @@ export default class RaidenService {
       if (!provider) {
         this.store.commit('noProvider');
       } else {
-        if (config) {
-          /* istanbul ignore next */
-          raiden = await RaidenService.createRaiden(
-            provider,
-            config.PRIVATE_KEY,
-            stateBackup,
-            subkey
-          );
-        } else {
-          // Check if trying to connect to main net
-          // and whether main net is allowed
-          /* istanbul ignore if */
-          if (
-            'networkVersion' in provider &&
-            provider.networkVersion === '1' &&
-            process.env.VUE_APP_ALLOW_MAINNET === 'false'
-          ) {
-            throw new RaidenError(ErrorCodes.RDN_UNRECOGNIZED_NETWORK);
-          }
-
-          raiden = await RaidenService.createRaiden(
-            provider,
-            undefined,
-            stateBackup,
-            subkey
-          );
+        // Check if trying to connect to main net
+        // and whether main net is allowed
+        /* istanbul ignore if */
+        if (
+          typeof provider !== 'string' &&
+          'networkVersion' in provider &&
+          provider.networkVersion === '1' &&
+          process.env.VUE_APP_ALLOW_MAINNET === 'false'
+        ) {
+          throw new RaidenError(ErrorCodes.RDN_UNRECOGNIZED_NETWORK);
         }
 
+        raiden = await RaidenService.createRaiden(
+          provider,
+          config?.PRIVATE_KEY ?? undefined,
```

Comment:
Doesn't `config?` already return `undefined`

Reply:
Yes this `??` is redundant basically but I missed it. Nice catch :D